corehq/apps/translations/integrations/transifex/views.py

```diff
@@ -296,6 +299,47 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+@method_decorator([toggles.APP_TRANSLATIONS_WITH_TRANSIFEX.required_decorator()], name='dispatch')
+class BlacklistTranslations(BaseTranslationsView):
+    page_title = _('Blacklist Translations')
+    urlname = 'blacklist_translations'
+    template_name = 'blacklist_translations.html'
+    section_name = ugettext_noop("Translations")
+
+    def section_url(self):
+        return self.page_url
+
+    @property
+    def blacklist_form(self):
+        if self.request.POST:
+            return AddTransifexBlacklistForm(self.domain, self.request.POST)
+        return AddTransifexBlacklistForm(self.domain)
+
+    @property
+    def page_context(self):
+        context = super(BlacklistTranslations, self).page_context
+        context['blacklisted_translations'] = self._blacklisted_translations
+        context['blacklist_form'] = self.blacklist_form
+        return context
+
+    def post(self, request, *args, **kwargs):
+        if self.transifex_integration_enabled(request):
+            if self.blacklist_form.is_valid():
+                self.blacklist_form.save()
+        return self.get(request, *args, **kwargs)
+
+    @property
+    def _blacklisted_translations(self):
+        blacklisted = TransifexBlacklist.objects.filter(domain=self.domain).all().values()
+        app_ids_to_name = {app.id: app.name for app in get_brief_apps_in_domain(self.domain)}
```

Comment:
i think you can add a method like [to_json](https://github.com/dimagi/commcare-hq/blob/9b750c2226b88bbeacd7238df3feb97e340208a9/corehq/apps/locations/models.py#L511) to `TransifexBlacklist` that does this conversion for you. I believe adding such parsing on the model is a better thing than adding it here so that it can be reused. There might be even better ways to add a parser(rails calls it presenter) to just handle this logic and not even model if you want to push for it.

Reply:
i meant to write for the logic that follows now to get the app_name in.

Reply:
https://github.com/dimagi/commcare-hq/pull/23429/commits/46627206cb30fa1144351950ccfbbbd71c033af2